WordPressKit/WordPressKit/ServiceRequest.swift

```diff
@@ -0,0 +1,70 @@
+import Foundation
+
+
+/// Enumeration to identify a service action
+///
+/// - subscribe: The action 'new'
+/// - unsubscribe: The action 'delete'
+/// - update: The action 'update'
+enum ServiceRequestAction: String {
+    case subscribe = "new"
+    case unsubscribe = "delete"
+    case update = "update"
+}
+
+
+/// A protocol for a Service request
+protocol ServiceRequest {
+    /// Returns a valid url path
+    var path: String { get }
+    
+    /// Returns the used API version
+    var apiVersion: ServiceRemoteWordPressComRESTApiVersion { get }
+}
+
+
+/// Reader Topic Service request
+///
+/// - notifications: Request type notifications
+/// - postsEmail: Posts email type notifications
+/// - comments: Comments type notifications
+enum ReaderTopicServiceSubscriptionsRequest {
+    case notifications(siteId: NSNumber, action: ServiceRequestAction)
+    case postsEmail(siteId: NSNumber, action: ServiceRequestAction)
+    case comments(siteId: NSNumber, action: ServiceRequestAction)
+    
+    
+    //MARK: Private methods
```

Comment:
Nipticky: Missing space between `//` and `MARK`

Reply:
+1